lib/Doctrine/ORM/Cache/DefaultEntityHydrator.php

```diff
@@ -73,7 +73,7 @@ public function __construct(EntityManagerInterface $em)
     public function buildCacheEntry(ClassMetadata $metadata, EntityCacheKey $key, $entity)
     {
         $data = $this->uow->getOriginalEntityData($entity);
-        $data = array_merge($data, $key->identifier); // why update has no identifier values ?
```

Comment:
@guilhermeblanco I suppose you mean in `Doctrine\ORM\Cache\Persister\EntityAbstractEntityPersister::loadById`? I'm not sure how to fix it in there. If you give me some hint how, I will try to fix the issue there.